src/widgets/immunizations/vaccination-row.component.tsx

```diff
@@ -0,0 +1,194 @@
+              </table>
+            </td>
+          </tr>
+        )}
+      </React.Fragment>
+    )
+  );
+}
+
+function getRecentVaccinationText(t, patientImmunization) {
+  if (!hasExistingDoses(patientImmunization)) {
+    return "";
+  }
+  let recentDose = patientImmunization.doses[0];
+  if (hasSequence(patientImmunization)) {
+    return (
+      recentDose.sequenceLabel +
+      " on " +
+      dayjs(recentDose.occurrenceDateTime).format("DD-MMM-YYYY")
+    );
+  }
+  const singleDoseText = t("single dose", "Single Dose");
+  return (
+    singleDoseText +
+    " on " +
+    dayjs(recentDose.occurrenceDateTime).format("DD-MMM-YYYY")
+  );
+}
+
+function hasExistingDoses(patientImmunization: any) {
+  return patientImmunization.doses && patientImmunization.doses.length > 0;
+}
+
+function hasSequence(patientImmunization) {
+  return (
+    patientImmunization?.sequences && patientImmunization?.sequences?.length > 0
+  );
+}
+
+function renderSequenceTable(match, t, immunization) {
+  return immunization?.doses?.map((dose, i) => {
+    return (
+      <tr key={`${immunization.uuid}-${i}`}>
+        {hasSequence(immunization) && <td>{dose.sequenceLabel}</td>}
+        {hasSequence(immunization) || (
+          <td>{t("single dose", "Single Dose")}</td>
+        )}
+        <td>
+          <div className={`${styles.alignRight}`}>
+            {dayjs(dose.occurrenceDateTime).format("DD-MMM-YYYY")}
+          </div>
+        </td>
+        <td>
+          <div className={`${styles.alignRight}`}>
+            {dayjs(dose.expirationDate).format("DD-MMM-YYYY")}
+          </div>
+        </td>
+        <td>
+          {
+            <Link to={`${match.path}/${dose.immunizationObsUuid}`}>
+              <svg
+                className="omrs-icon"
+                fill="var(--omrs-color-ink-low-contrast)"
+                onClick={() =>
+                  openWorkspaceTab(ImmunizationsForm, "Immunizations Form", [
+                    {
+                      vaccineName: immunization?.vaccineName,
+                      vaccineUuid: immunization?.vaccineUuid,
+                      immunizationObsUuid: dose?.immunizationObsUuid,
+                      encounterUuid: dose?.encounterUuid,
+                      manufacturer: dose.manufacturer.display,
+                      lotNumber: dose.lotNumber,
+                      expirationDate: dose.expirationDate,
+                      sequences: immunization.sequences,
+                      currentDose: {
+                        sequenceLabel: dose.sequenceLabel,
+                        sequenceNumber: dose.sequenceNumber
+                      },
+                      vaccinationDate: dose.occurrenceDateTime
+                    }
+                  ])
+                }
+              >
+                <use xlinkHref="#omrs-icon-chevron-right" />
+              </svg>
+            </Link>
+          }
+        </td>
+      </tr>
+    );
+  });
+}
+
+const capitalize = s => {
```

Comment:
We could use the same function from `lodash`. I think `lodash` is anyway used in the lib, right?

Reply:
Otherwise, please type it properly ;)